### ISSUE TITLE
`validate` throws an exception if option is unset

### ISSUE THREAD

**Issue by jkinkead** [scallop/scallop#108]
If you have a `validate` function defined for an option, it causes the option to be dereferenced (`get` called on it) regardless of whether its set or not. This results in the following exception:

``` scala
java.util.NoSuchElementException: None.get
    at scala.None$.get(Option.scala:347)
    at scala.None$.get(Option.scala:345)
    at org.rogach.scallop.ScallopOption.apply(ScallopOption.scala:26)
    at org.rogach.scallop.ScallopConfValidations$$anonfun$validate$1.apply(ScallopConfValidations.scala:19)
    at org.rogach.scallop.ScallopConfValidations$$anonfun$validate$1.apply(ScallopConfValidations.scala:18)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$1.apply(ScallopConf.scala:361)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$1.apply(ScallopConf.scala:360)
    at scala.collection.immutable.List.foreach(List.scala:381)
    at org.rogach.scallop.ScallopConf.runValidations(ScallopConf.scala:360)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$2$$anonfun$apply$7.apply(ScallopConf.scala:371)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$2$$anonfun$apply$7.apply(ScallopConf.scala:369)
    at scala.Option.foreach(Option.scala:257)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$2.apply(ScallopConf.scala:369)
    at org.rogach.scallop.ScallopConf$$anonfun$runValidations$2.apply(ScallopConf.scala:368)
    at scala.Option.foreach(Option.scala:257)
    at org.rogach.scallop.ScallopConf.runValidations(ScallopConf.scala:368)
    at org.rogach.scallop.ScallopConf.verifyBuilder(ScallopConf.scala:352)
    at org.rogach.scallop.ScallopConf.verify(ScallopConf.scala:595)
```

It seems like `validate` should only be called on set values, since the `required` value enforces if it's set or not.

**Comment by jkinkead** [comment]
This only happens when all arguments to a `validate` function are not supplied. If a mixture are supplied, a codependency failure is raised instead.

It might be nice to support validation of multiple arguments, some with defaults; that is, change the codependency addition to a "all have non-empty value". There's a lot of situations where you'd want to validate the relation of two options, which both may have defaults:

``` scala
object Conf extends ScallopConf(List("-a","15")) {
  val start = opt[Int]("start", default = Some(1))
  val end = opt[Int]("end", default = Some(10))
  // Does not work - defaults will never be used.
  validate (start, end) { (s,e) =>
    if (start <= end) Right(Unit)
    else Left("Start must be before end")
  }
}
```

**Comment by Rogach** [comment]
I pushed the fix - now your cases should work as expected. If everything is ok, I will release new version soon.

**Comment by jkinkead** [comment]
Awesome, that looks great! Thanks for the quick turn-around. :)

**Comment by Rogach** [comment]
Released version 1.0.2, should sync to Maven Central within an hour.
